src/jdk.httpserver/share/classes/sun/net/httpserver/simpleserver/FileServerHandler.java

```diff
@@ -159,7 +159,7 @@ private void handleNotFound(HttpExchange exchange) throws IOException {
 
     private static void discardRequestBody(HttpExchange exchange) throws IOException {
         try (InputStream is = exchange.getRequestBody()) {
-            is.readAllBytes();
+            is.skip(Integer.MAX_VALUE);
```

Comment:
Hello Darragh, since the goal is to read all bytes of the InputStream and then close it, I wonder if we should instead do:

```java
try (InputStream is = exchange.getRequestBody()) {
    boolean eof = false;
    while (!eof) {
       is.skip(Long.MAX_VALUE);
       // check if stream has reached EOF
       final int nextByte = is.read();
       if (nextByte < 0) {
           eof = true;
       } 
     }

}
```
(We could have just called `while ( !eof) { eof = is.drain(Long.MAX_VALUE); }`, but drain() is a method on an internal implementation of the `InputStream`.)

In its current form (even before the changes proposed in the PR), the use of `readAllBytes()` doesn't guarantee that the entire stream is read. That can then mean that we might close the (socket) input stream before reading the entire body.

Reply:
IIRC the drain() method will be called if the input stream is closed without having read all the bytes, and subsequently  the connection will get closed if the stream is not EOF after drain(); So I believe what Darragh has is actually good enough. It's a bit of belt and braces already.